src/ir/module-utils.h

```diff
@@ -388,21 +388,22 @@ template<typename T> struct CallGraphPropertyAnalysis {
 
   // Propagate a property from a function to those that call it.
   //
-  // hasProperty() - Check if the property is present.
+  // hasProperty() - Check if the property is present. The second parameter is
+  //                 the function due to which we are adding the property, which
+  //                 may be useful for logging.
   // canHaveProperty() - Check if the property could be present.
-  // addProperty() - Adds the property. This receives a second parameter which
-  //                 is the function due to which we are adding the property.
-  void propagateBack(std::function<bool(const T&)> hasProperty,
+  // addProperty() - Adds the property.
+  void propagateBack(std::function<bool(const T&, Function*)> hasProperty,
                      std::function<bool(const T&)> canHaveProperty,
-                     std::function<void(T&, Function*)> addProperty,
+                     std::function<void(T&)> addProperty,
                      NonDirectCalls nonDirectCalls) {
     // The work queue contains items we just learned can change the state.
     UniqueDeferredQueue<Function*> work;
     for (auto& func : wasm.functions) {
       if (hasProperty(map[func.get()]) ||
```

Comment:
Wait, no, that doesn't work for this call. What can we do here? Maybe pass NULL and check that in the logging code?